staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go

```diff
@@ -615,7 +615,7 @@ func (c *Cacher) Watch(ctx context.Context, key string, opts storage.ListOptions
 	// to compute watcher.forget function (which has to happen under lock).
 	watcher := newCacheWatcher(
 		chanSize,
-		filterWithAttrsFunction(key, pred),
+		filterWithAttrsAndPrefixFunction(key, pred),
```

Comment:
some changed introduced in this commit could be moved to a new PR (refactoring). I can do that if that would be easier for review.

Reply:
It's fine, it's fairly simple and not that large. Let's keep as is.